Controllers/PurchasesController.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using StoresManagement.Data;
+using StoresManagement.Models;
+using StoresManagement.ViewModels;
+
+namespace StoresManagement.Controllers
+{
+    public class PurchasesController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public PurchasesController(ApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        // GET: Purchases
+        public async Task<IActionResult> Index()
+        {
+            var Purchases = await _context.Purchases
+                .Include(b => b.Branch)
+                .Include(b => b.Branch.Entity)
+                .Include(b => b.Customer)
+                .ToListAsync();
+
+            var branchesVM = new List<PurchaseFormViewModel>();
+
+            foreach (var purchase in Purchases)
+            {
+                branchesVM.Add(_mapper.Map<PurchaseFormViewModel>(purchase));
+            }
+
+            return View(branchesVM);
+        }
+
+        // GET: Purchase by Product Id
+        public async Task<IActionResult> ListPurchases(int? id)
+        {
+            var Purchases = await _context.Purchases
+                .Include(b => b.Branch)
+                .Include(b => b.Branch.Entity)
+                .Include(b => b.Customer)
+                .Where(m => m.BranchId == id || m.CustomerId == id)
```

Comment:
A Purchase is related to a Branch and a Customer.
From the Branch page, you can browse the list of the purchases for a dedicated branch.
From the Customer page, you can browse the list of the purchases for a dedicated customer.

I am using the same view (List of purchases) for Branch/Customer. 
This was the solution for retrieving the list of purchases related to Branch or Customer.

**DEFECT -  TO BE CHANGED**: It will bring the list of purchases for branch and customer with same id.

Reply:
Ok now I got your intention. It is still code smell.
To clear this I would provide separete endpoints.